### PR TITLE
[CHI-270] Responsive fluid buttons

### PR DESCRIPTION
**Pull request by davidojedacl**
@mattnickles @jllr

### CODE REVIEW

File: src/website/views/templates/authentication.pug [hunk 1]
```diff
@@ -27,21 +27,22 @@ h3 App Basic Sign In
...
-                label(for='email') Email
-              .a-inputWrapper.-floatingLabel.-pt--1#apassword
-                input(type='password', class='a-input -xl', id='password')
-                label(for='password') Password
-              .m-formGroup.-pt--3.-pt-sm--4.-justify-content--between.-align-items--center.-flex--column.-flex-sm--row
-                a.-text.-mb--2.-mb-sm--0(href="#") Forgot email or password?
-                button.a-btn.-primary.-xl Sign In
-          .-text.-text--center.-py--2 Don't have an account? <a href="#">Sign Up</a>
+      .chi-app__body.-px-sm--2.-py-sm--4
+        .a-grid.-pb--2.-justify-content--center
```
Comment (jllr) [inline src/website/views/templates/authentication.pug]: @davidojedacl please remove .a-grid and .-justify-content--center and add .-w--100 .-mw--480 .-m--auto

File: src/website/views/templates/authentication.pug [hunk 2]
```diff
@@ -27,21 +27,22 @@ h3 App Basic Sign In
...
+          .a-col.-w-md--7
```
Comment (jllr) [inline src/website/views/templates/authentication.pug]: @davidojedacl this div and classes are not necessary anymore, please remove it

File: src/website/views/templates/authentication.pug [hunk 3]
```diff
@@ -27,21 +27,22 @@ h3 App Basic Sign In
...
+            .a-card.-b--0.-bg--none
+              .a-card__content.-text--center.-px--4.-px-sm--6.-pb--4
```
Comment (jllr) [inline src/website/views/templates/authentication.pug]: @davidojedacl please remove .-text--center

File: src/website/views/templates/authentication.pug [hunk 4]
```diff
@@ -27,21 +27,22 @@ h3 App Basic Sign In
...
+                h3.-text--normal Sign In
```
Comment (jllr) [inline src/website/views/templates/authentication.pug]: @davidojedacl add .-text--center

File: src/website/views/templates/authentication.pug [hunk 5]
```diff
@@ -27,21 +27,22 @@ h3 App Basic Sign In
...
+                .a-inputWrapper.-floatingLabel.-pt--2#aemail
+                  input(type='email', class='a-input -xl', id='email')
+                  label(for='email') Email
+                .a-inputWrapper.-floatingLabel.-pt--1#apassword
+                  input(type='password', class='a-input -xl', id='password')
+                  label(for='password') Password
+                .m-formGroup.-pt--3.-pt-sm--4.-justify-content--between.-align-items--center.-flex--column.-flex-sm--row
```
Comment (jllr) [inline src/website/views/templates/authentication.pug]: @davidojedacl please remove .m-formGroup, .-justify-content--between, .-align-items--center and .-flex-sm--row.
And add .-d--flex, .-flex--column and .-flex-md--row

File: src/website/views/templates/authentication.pug [hunk 6]
```diff
@@ -27,21 +27,22 @@ h3 App Basic Sign In
...
+                  a.-text.-mb--2.-mb-sm--0(href="#") Forgot email or password?
```
Comment (jllr) [inline src/website/views/templates/authentication.pug]: @davidojedacl please remove .-mb-sm--0 and add .-d--flex, .-justify-content--center, .-justify-content-md--start, .-align-items-md--center, .a-col and .-w-md--9

File: src/website/views/templates/authentication.pug [hunk 7]
```diff
@@ -27,21 +27,22 @@ h3 App Basic Sign In
...
+                  button.a-btn.-primary.-xl Sign In
```
Comment (jllr) [inline src/website/views/templates/authentication.pug]: @davidojedacl please add .-fluid, .-justify-content--center, .a-col and .-w-md--3

File: src/chi/components/header/header.scss [hunk 8]
```diff
@@ -222,10 +222,3 @@
     }
   }
 }
-
-/* IE11 CSS hack */
-@media all and (-ms-high-contrast:none) {
-  .o-header__content, .o-header__toolbar ~ .o-header__content {
-    height: 4rem;
```
Comment (jllr) [inline src/chi/components/header/header.scss]: @davidojedacl removing this classes header is not looking good in IE11. We must make these classes bigger, via media queries, and ensuring the smallest scenarios are covered with the new heights.

Screen Shot 2019-05-23 at 12 38 42 PM

Screen Shot 2019-05-23 at 12 39 03 PM

File: src/website/views/templates/authentication.pug [hunk 9]
```diff
@@ -27,21 +27,22 @@ h3 App Basic Sign In
...
-                input(type='password', class='a-input -xl', id='password')
-                label(for='password') Password
-              .m-formGroup.-pt--3.-pt-sm--4.-justify-content--between.-align-items--center.-flex--column.-flex-sm--row
-                a.-text.-mb--2.-mb-sm--0(href="#") Forgot email or password?
-                button.a-btn.-primary.-xl Sign In
-          .-text.-text--center.-py--2 Don't have an account? <a href="#">Sign Up</a>
+      .chi-app__body.-px-sm--2.-py-sm--4
+        .a-grid.-pb--2.-justify-content--center
+          .a-col.-w-md--7
+            .a-card.-b--0.-bg--none
```
Comment (jllr) [inline src/website/views/templates/authentication.pug]: @davidojedacl please add .a-grid

**Comment by davidojedacl** [comment]
I'll change it but I think it's WRONG markup. It should not be done like that. I think you should not use a-col in html elements that are content like buttons or anchor, nor a-grid in elements like card that should be content. the right markup is to use a-grid and a-col separatly as indicated in the documentation.

Screen Shot 2019-05-27 at 12 17 28 PM

**Comment by davidojedacl** [comment]
Pending IE11 header support

File: src/website/views/templates/authentication.pug [hunk 10]
```diff
@@ -27,21 +27,29 @@ h3 App Basic Sign In
...
+                    .a-inputWrapper.-floatingLabel.-pt--2#aemail
+                      input(type='email', class='a-input -xl', id='email')
+                      label(for='email') Email
+                  div.a-col.-w--12.-mb--3.-mb-sm--4
+                    .a-inputWrapper.-floatingLabel.-pt--1#apassword
+                      input(type='password', class='a-input -xl', id='password')
+                      label(for='password') Password
+                  div.a-col.-w-md--9.-mb--2.-mb-sm--0
+                    a.-text.-mb-sm--0.-d--flex.-justify-content--center.-justify-content-md--start(href="#") Forgot email or password?
+                  div.a-col.-w-md--3
```
Comment (jllr) [inline src/website/views/templates/authentication.pug]: @davidojedacl can you change the col from .-w-md--3 to .-w-sm--3?

File: src/website/views/templates/authentication.pug [hunk 11]
```diff
@@ -27,21 +27,29 @@ h3 App Basic Sign In
...
+                  div.a-col.-w--12
+                    .a-inputWrapper.-floatingLabel.-pt--2#aemail
+                      input(type='email', class='a-input -xl', id='email')
+                      label(for='email') Email
+                  div.a-col.-w--12.-mb--3.-mb-sm--4
+                    .a-inputWrapper.-floatingLabel.-pt--1#apassword
+                      input(type='password', class='a-input -xl', id='password')
+                      label(for='password') Password
+                  div.a-col.-w-md--9.-mb--2.-mb-sm--0
+                    a.-text.-mb-sm--0.-d--flex.-justify-content--center.-justify-content-md--start(href="#") Forgot email or password?
```
Comment (jllr) [inline src/website/views/templates/authentication.pug]: @davidojedacl can you change the col from .-justify-content-md--start to .-justify-content-sm--start?

File: src/website/views/templates/authentication.pug [hunk 12]
```diff
@@ -63,26 +71,46 @@ h3 App Basic Sign In
...
-              <div class="m-formGroup -justify-content--between -align-items--center -flex--column -flex-sm--row -pt--3 -pt-sm--4">
-                <a class="-text -mb--2 -mb-sm--0" href="#">Forgot email or password?</a>
-                <button class="a-btn -primary -xl">Sign In</button>
-              </div>
+            <div class="a-card__content -px--4 -px-sm--6">
+              <fieldset>
+                <div class="a-grid -noGutter -justify-content--center -align-items--center">
+                  <div class="a-col -w--12">
+                    <h3 class="-text--normal -text--center">Sign In</h3>
+                    <div class="m-alert -danger -center" role="alert">
```
Comment (jllr) [inline src/website/views/templates/authentication.pug]: @davidojedacl this must be removed

File: src/chi/components/header/header.scss [hunk 13]
```diff
@@ -224,7 +224,13 @@
 }
 
 /* IE11 CSS hack */
-@media all and (-ms-high-contrast:none) {
+@media only screen and (max-width: $lg-breakpoint - 1) and (-ms-high-contrast:none) {
```
Comment (jllr) [inline src/chi/components/header/header.scss]: @davidojedacl this is how I see in IE11. We should make it just for sm or md devices

Screen Shot 2019-05-28 at 4 13 14 PM

Screen Shot 2019-05-28 at 4 12 25 PM

File: src/website/views/templates/authentication.pug [hunk 14]
```diff
@@ -27,21 +27,29 @@ h3 App Basic Sign In
...
+                    h3.-text--normal.-text--center Sign In
+                  div.a-col.-w--12
+                    .a-inputWrapper.-floatingLabel.-pt--2#aemail
+                      input(type='email', class='a-input -xl', id='email')
+                      label(for='email') Email
+                  div.a-col.-w--12.-mb--3.-mb-sm--4
+                    .a-inputWrapper.-floatingLabel.-pt--1#apassword
+                      input(type='password', class='a-input -xl', id='password')
+                      label(for='password') Password
+                  div.a-col.-w-md--9.-mb--2.-mb-sm--0
```
[the remark this answers is not in the document]
Comment (davidojedacl) [reply src/website/views/templates/authentication.pug]: done

File: src/website/views/templates/authentication.pug (hunk 10 above)
Comment (davidojedacl) [reply src/website/views/templates/authentication.pug]: done

File: src/website/views/templates/authentication.pug (hunk 11 above)
Comment (davidojedacl) [reply src/website/views/templates/authentication.pug]: No, we must leave  .-justify-content-md--start

File: src/website/views/templates/authentication.pug (hunk 12 above)
Comment (davidojedacl) [reply src/website/views/templates/authentication.pug]: done

File: src/chi/components/header/header.scss (hunk 13 above)
Comment (davidojedacl) [reply src/chi/components/header/header.scss]: done

File: src/website/views/templates/authentication.pug [hunk 15]
```diff
@@ -27,21 +27,29 @@ h3 App Basic Sign In
...
+                  div.a-col.-w--12
+                    .a-inputWrapper.-floatingLabel.-pt--2#aemail
+                      input(type='email', class='a-input -xl', id='email')
+                      label(for='email') Email
+                  div.a-col.-w--12.-mb--3.-mb-sm--4
+                    .a-inputWrapper.-floatingLabel.-pt--1#apassword
+                      input(type='password', class='a-input -xl', id='password')
+                      label(for='password') Password
+                  div.a-col.-w-sm--12.-w-md--9.-mb--2.-mb-md--0
+                    a.-text.-mb-sm--0.-d--flex.-justify-content--center.-justify-content-md--start(href="#") Forgot email or password?
```
Comment (jllr) [inline src/website/views/templates/authentication.pug]: @davidojedacl can you remove .-mb-sm--0 as it is not necessary?

File: src/website/views/templates/authentication.pug [hunk 16]
```diff
@@ -27,21 +27,29 @@ h3 App Basic Sign In
...
+            .a-card__content.-px--4.-px-sm--6
+              fieldset
+                .a-grid.-noGutter.-justify-content--center.-align-items--center
+                  div.a-col.-w--12
+                    h3.-text--normal.-text--center Sign In
+                  div.a-col.-w--12
+                    .a-inputWrapper.-floatingLabel.-pt--2#aemail
+                      input(type='email', class='a-input -xl', id='email')
+                      label(for='email') Email
+                  div.a-col.-w--12.-mb--3.-mb-sm--4
```
Comment (jllr) [inline src/website/views/templates/authentication.pug]: @davidojedacl @mattnickles just a doubt on these margins .-mb-sm--4 should be .-mb-md--(n) or should we removed and just let .-mb--3? Thoughts?

File: src/website/views/templates/authentication.pug (hunk 15 above)
Comment (davidojedacl) [reply src/website/views/templates/authentication.pug]: done

**Pull request closed without merging**